x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwkSetLoader.java

```diff
@@ -101,7 +101,7 @@ private ListenableFuture<Boolean> getFuture() {
 
             final ListenableFuture<Boolean> newFuture = new ListenableFuture<>();
             if (this.reloadFutureRef.compareAndSet(null, newFuture)) {
-                loadInternal(ActionListener.runAfter(newFuture, () -> {
+                loadInternal(ActionListener.runBefore(newFuture, () -> {
```

Comment:
Err this is not the right fix. It potentially turns a test issue into a production one. Having the `reloadFutureRef` reset before the listener is invoked breaks the guarantee that `loadInternal` is single threaded. I think I should just fix for the test failure and avoid changing production code. I'll split this into a separate PR and leave this one to just fix the connection request timeout.